confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -366,67 +372,62 @@ def test_compatibility(self, subject, avro_schema, version='latest'):
         try:
             result, code = self._send_request(url, method='POST', body=body)
             if code == 404:
-                log.error(("Subject or version not found:" + str(code)))
+                log.error("Subject or version not found. Error code: {}".format(code))
                 return False
             elif code == 422:
-                log.error(("Invalid subject or schema:" + str(code)))
+                log.error("Invalid subject or schema. Error code: {}".format(code))
                 return False
-            elif code >= 200 and code <= 299:
+            elif 200 <= code <= 299:
                 return result.get('is_compatible')
             else:
-                log.error("Unable to check the compatibility: " + str(code))
+                log.error("Unable to check the compatibility. Error code: {}".format(code))
                 return False
         except Exception as e:
             log.error("_send_request() failed: %s", e)
             return False
 
-    def update_compatibility(self, level, subject=None):
```

Comment:
Setting per subject compatibility is part of the Schema Registry API and should be maintained. 

https://docs.confluent.io/current/schema-registry/docs/api.html#put--config-(string-%20subject)

Reply:
my mistake, I must have missed that. Do you think it should revert to the global compatibility level if there's a `404` response?

Reply:
I don't think it would be a good idea edit the compatibility for all other subjects if just one subject has a 404

Reply:
@cricket007 Sorry you're right I meant to write this on `get_compatibility` - thoughts on that?

Reply:
I think the spec says that the 404 corresponds to the subject not being found. Personally, I think it is a good idea to return the global compatibility as a fallback, but I don't think the Java clients do that (yet)?

Reply:
Ok. I'll stay safe and leave as-is for now, unless there's a specific need to make it behave otherwise